extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/BridgeHandler.java

```diff
@@ -0,0 +1,643 @@
+                }
+                sceneListener = null;
+            }
+
+            if (connMan.checkConnection()) {
+                devStatMan.start();
+            }
+
+            boolean configChanged = false;
+            if (connMan.getApplicationToken() != null) {
+                configuration.remove(USER_NAME);
+                configuration.remove(PASSWORD);
+                logger.debug(connMan.getApplicationToken());
+                configuration.put(APPLICATION_TOKEN, connMan.getApplicationToken());
+                configChanged = true;
+            }
+            String dSSname = configuration.get(DigitalSTROMBindingConstants.DS_NAME).toString();
+            if ((dSSname == null || (dSSname != null && dSSname.isEmpty())) && connMan.checkConnection()) {
+                dSSname = connMan.getDigitalSTROMAPI().getInstalationName(connMan.getSessionToken());
+
+                if (dSSname != null) {
+                    configuration.put(DS_NAME, dSSname);
+                    configChanged = true;
+                }
+            }
+            if (configChanged) {
+                updateConfiguration(configuration);
+            }
+        }
+    };
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing DigitalSTROM Bridge handler.");
+
+        Configuration configuration = getConfig();
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Checking configuration...");
+        if (!(configuration.get(HOST) != null && !configuration.get(HOST).toString().isEmpty())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "The connection to the digitalSTROM-Server can't established, because the host is missing. Please set the host.");
+            return;
+        }
+
+        String[] loginConfig = getLoginConfig(configuration);
+        if (!checkLoginConfig(loginConfig)) {
+            return;
+        }
+
+        // Start an extra thread to readout the configuration and check the connection, because it takes sometimes more
+        // than 5000 milliseconds and the handler will suspend (ThingStatus.UNINITIALIZED).
+        scheduler.execute(new InitalizazeRunnable(this, configuration, loginConfig));
+
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+
+        if (devStatMan != null) {
+            devStatMan.unregisterTotalPowerConsumptionListener();
+            devStatMan.unregisterStatusListener();
+            this.devStatMan.stop();
+        }
+        if (connMan != null) {
+            connMan.unregisterConnectionListener();
+        }
+        if (this.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // nothing to do
```

Comment:
you should handle the REFRESH command, which let's the user trigger a refresh of the state of a certain channel.